src/Arcus.Security.Providers.AzureKeyVault/Extensions/SecretStoreBuilderExtensions.cs

```diff
@@ -500,6 +553,29 @@ public static class SecretStoreBuilderExtensions
                 configureOptions);
         }
 
+        /// <summary>
+        /// Adds Azure Key Vault as a secret source which uses Managed Identity authentication.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="rawVaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
+        /// <param name="cacheConfiguration">The configuration to control how the caching will be done.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="rawVaultUri"/> is blank.</exception>
+        public static SecretStoreBuilder AddAzureKeyVaultWithManagedIdentity(
+            this SecretStoreBuilder builder,
+            string rawVaultUri,
+            ICacheConfiguration cacheConfiguration)
```

Comment:
I think it is sometimes confusing to have multiple overloads which actually offer the same functionality, but do it in a different way.

I am referring to this method which takes a `cacheConfiguration` argument, and the method that has an `allowCaching` parameter.
Maybe we should get rid of the method that takes an `allowCaching` parameter; if you want to have caching and are happy with the defaults, you could introduce `CacheConfiguration.Default` which you can pass to this method:

```csharp
builder.AddAzureKeyVaultWithManagedIdentity("myvaulturi", CacheConfiguration.Default);
```

Reply:
Then, I think you could limit yourself to just one method for adding KeyVault using MSI, more specifically the one method that takes all the possible arguments, and specify defaults:

```csharp
AddAzureKeyVaultWithManagedIdentity(
    this SecretStoreBuilder builder,
    string vaultUri, // this one is mandatory offcourse
    string name = null,
    ICacheConfiguration cacheConfiguration = null, // no caching
    Func<string, string> mutateSecretName = null,
    Action<KeyVaultOptions> configureOptions);
```
Or maybe 2 methods.  You could just specify the overload which takes the one mandatory parameter as well, just for the sake of clarity ?

Reply:
> I think it is sometimes confusing to have multiple overloads which actually offer the same functionality, but do it in a different way.
> I am referring to this method which takes a cacheConfiguration argument, and the method that has an allowCaching parameter.
Maybe we should get rid of the method that takes an allowCaching parameter; if you want to have caching and are happy with the defaults, you could introduce CacheConfiguration.Default which you can pass to this method:

Yes, I see your point, but this is a suggestion outside the purpose of this PR. It's more a general concern since _all_ the overloads work like this (combination of `allowCaching` and `ICacheConfiguration`).

About the suggestion to make it optional parameters, I strongly suggest we don't do that bc that was the sole reason we had problems in the past with introducing new methods, signature conflicts, `MethodNotFound` failures... all bc it's harder to maintain to have optional parameters.
So, we've chosen to have method overloads with more or less parameters to choose from instead of one big method with almost everything optional.

Reply:
I can agree with having multiple overloads instead of optionals in a library.  However, it might be fruitfull to re-think the overloads that are now being introduced to make sure that the most common scenario's are being covered without having to specify unnecessary parameters.
For instance, I think (now I'm speaking for myself, from my POV) the mutate-secretname parameter is one that is required in most cirumstances.  Also caching is often required (default would do).  I also wonder what is going on internally when you do not specify a name, does the library assign a name to the provider itself ?  (In that case, I do think name might be created as an optional argument, just like the allowCaching parameter is).

Reply:
> it might be fruitfull to re-think the overloads that are now being introduced to make sure that the most common scenario's are being covered without having to specify unnecessary parameters.

Yes of course! Lets discuss this in a discussion/issue to see where we can go to. Having a default caching configuration instead of a boolean is something for example I'm all for.

> I also wonder what is going on internally when you do not specify a name, does the library assign a name to the provider itself ? 

If there's no name specified, then there's no name available internally too. It would be a kind off wrong design (IMO) to set a default name internally. Only 'named' secret secret providers which the consumer has configured, will have a name and is available when retrieving via the secret store.
And, yes, I can see the usefulness of optional parameters, but we try to limit here from using it bc it's a very hard thing to maintain. Also, we have created overloads with minimal and maximium parameters to circumvent this problem. 
But this can be discussed too in a different issue/disscussion.